docs/entities/data.md

```diff
@@ -0,0 +1,98 @@
+---
+sidebar_position: 2
+---
+# Data and Networking
+
+An entity without data is quite useless, as such storing data on an entity is essential. All entities store some default data, such as their type and their position. This article will explain how to add your own data, as well as how to synchronize that data.
+
+The most simple way to add data is as a field in your `Entity` class. You can then interact with this data in any way you wish. However, this quickly becomes very annoying as soon as you have to synchronize that data. This is because most entity logic is run on the server only, and it is only occasionally (depending on the [`EntityType`][entitytype]'s `clientUpdateInterval` value) that an update is sent to the client; this is also the cause for easily noticeable entity "lags" when the server's tick speed is too slow.
+
+As such, vanilla introduces a few systems to help with that, each of which serves a specific purpose.
+
+## `SynchedEntityData`
+
+`SynchedEntityData` is a system used for storing values at runtime and syncing them over the network. It is split into three classes:
+
+- `EntityDataSerializer`s are basically wrappers around a [`StreamCodec`][streamcodec].
+    - Minecraft uses a hard-coded map of serializers. NeoForge transforms this map into a registry, meaning that if you want to add new `EntityDataSerializer`s, they must be added by [registration].
+    - Minecraft defines various default `EntityDataSerializer`s in the `EntityDataSerializers` class.
+- `EntityDataAccessor`s are held by the entity and are used to get and set the data values.
+- `SynchedEntityData` itself holds all `EntityDataAccessor`s for an entity, and automatically calls on the `EntityDataSerializer`s to sync values as needed.
+
+To get started, create an `EntityDataAccessor` in your entity class:
+
+```java
+public class MyEntity extends Entity {
+    // The generic type must match the one of the second parameter below.
+    public static final EntityDataAccessor<Integer> MY_DATA =
+        SynchedEntityData.defineId(
+            // The class of the entity.
+            MyEntity.class,
+            // The entity data accessor type.
+            EntityDataSerializers.INT
+        );
+}
+```
+
+:::danger
+While the compiler will allow you to use a class other than the owning class as the first parameter in `SynchedEntityData#defineId()`, this can and will lead to hard-to-debug issues and as such should be avoided.
```

Comment:
I personally prefer the wording used in the old entity networking page. This is not a case of "avoid if possible", it's a case of "seriously, don't do it, it WILL break".
I would also add a note that this includes adding data accessors via mixin to the *technically* correct class.